vscode/src/jsonrpc/agent-protocol.ts

```diff
@@ -630,3 +633,13 @@ export interface CreateFilesParams {
 export interface RenameFilesParams {
     files: { oldUri: string; newUri: string }[]
 }
+
+export type CustomCommandResult = CustomChatCommandResult | CustomEditCommandResult
+export interface CustomChatCommandResult {
+    type: 'chat'
+    result: string
```

Comment:
One more blocking comment: using different types for the `result` property is also problematic. It's better to use unique property names for every unique type
```suggestion
    panelID: string
```

BTW, I am working on a code generation tool that automatically catches these errors and will fail CI with an error message on how to fix the problem

Reply:
I'm not sure if `panelID` is the best property name, just added a suggestion to give an example of a diff that would resolve the issue. We encode discriminated unions as flat data classes in other languages. For example:

```
data class CustomCommandResult(
    type: String,
    chatResult: String? = null,
    editResult: EditTask? = null,
)
```